validator/accounts/v2/wallet_recover.go

```diff
@@ -0,0 +1,131 @@
+package v2
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/manifoldco/promptui"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/validator/flags"
+	v2keymanager "github.com/prysmaticlabs/prysm/validator/keymanager/v2"
+	"github.com/prysmaticlabs/prysm/validator/keymanager/v2/derived"
+	"github.com/urfave/cli/v2"
+)
+
+const phraseWordCount = 24
+
+func RecoverWallet(cliCtx *cli.Context) error {
+	// Read a wallet's directory from user input.
+	walletDir, err := inputWalletDir(cliCtx)
+	if err != nil && !errors.Is(err, ErrNoWalletFound) {
+		log.Fatalf("Could not parse wallet directory: %v", err)
+	}
+
+	// Check if the user has a wallet at the specified path.
+	// If a user does not have a wallet, we instantiate one
+	// based on specified options.
+	walletExists, err := hasDir(walletDir)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if walletExists {
+		log.Fatal(
+			"You already have a wallet at the specified path. You can " +
+				"edit your wallet configuration by running ./prysm.sh validator wallet-v2 edit",
+		)
+	}
+	if err = recoverDerivedWallet(cliCtx, walletDir); err != nil {
```

Comment:
We need to make sure this is a derived wallet in the first place

Reply:
Theres no existing wallet if you're using recover no? The function already rejects if a wallet exists.